### PR TITLE
(EAI-498): Upgrade main responder LLM + system prompt to reduce AI non answers

### PR DESCRIPTION
**Pull request by mongodben**
Jira: https://jira.mongodb.org/browse/EAI-498

## Changes

Based on the changes in #482, the production application is providing non-answers to too many questions b/c the exact answer isn't in the context information. This PR:

- Upgrades the main responder LLM from `gpt-4o-mini` to `gpt-4o`. 
- Updates the system prompt to allow greater freedom for the LLM to respond based on it's internal knowledge.
  - We know that GPT-4o has much better internal knowledge of MongoDB based on our benchmarking of LLMs for MongoDB + much better general reasoning ability based on common benchmarks than GPT-3.5-turbo, our previous LLM, this should be a relatively safe move. While it might introduce some more hallucinations in various edge cases, it should have a net-positive effect.
- In k8s, pass models as non-secret env vars instead of k8s secrets to allow for easier changing.

## Notes

- Evaluation results: https://www.braintrust.dev/app/mdb-test/p/mongodb-chatbot-conversations/experiments/gpt-35-preprocessor-gpt4o-responder-1724684610280
  - Note the increase in AnswerRelevancy and Faithfulness compared to `
gpt-35-preprocessor-gpt4o-mini-responder-1724088921632` and increase in Faithfulness compared to `gpt-35-1724081121844`

### CODE REVIEW

File: packages/chatbot-eval-mongodb-public/src/braintrust/evaluateRagConversationsData.ts [hunk 1]
```diff
@@ -21,11 +21,17 @@ const runs = [
...
+  //     model: "gpt-35-preprocessor-gpt4o-mini-responder",
+  //     generatedDataRunId: new Object("66c37e6c97b47998dd59e87e").toString(),
+  //     description: `Uses GPT-3.5 for pre-processsing tasks: metadata extraction, user input guardrail, and query rewriting.
+  // Uses GPT-4o mini for the main responder.`,
+  //   },
   {
-    model: "gpt-35-preprocessor-gpt4o-mini-responder",
-    generatedDataRunId: new Object("66c37e6c97b47998dd59e87e").toString(),
+    model: "gpt-35-preprocessor-gpt4o-responder",
+    generatedDataRunId: new Object("66cc95022525d293a0ef4cc3").toString(),
```
Comment (nlarew) [inline packages/chatbot-eval-mongodb-public/src/braintrust/evaluateRagConversationsData.ts:32]:
```suggestion
    generatedDataRunId: new ObjectId("66cc95022525d293a0ef4cc3").toString(),
```
I think this is an issue throughout this file - fun little quirk of JS that this works currently!

File: packages/chatbot-server-mongodb-public/src/systemPrompt.ts [hunk 2]
```diff
@@ -1,18 +1,23 @@
...
+Be concise and informative in your responses.
 You were created by MongoDB.
-Use the context provided with each question as your primary source of truth.
-If you do not know the answer to the question based on the provided documentation content, respond with the following text:
+Use the provided context information to answer user questions. You can also use your internal knowledge of MongoDB to inform the answer.
+
+If you do not know the answer to the question, respond with the following text:
 "I'm sorry, I do not know how to answer that question. Please try to rephrase your query."
-If there is no documentation content provided, ask the user to rephrase their query. Provide a few suggestions for how to rephrase the query.
+Also ask the user to rephrase their query. Provide a few suggestions for how to rephrase the query.
```
Comment (nlarew) [inline packages/chatbot-server-mongodb-public/src/systemPrompt.ts:14]: > Provide a few suggestions for how to rephrase the query.

Might be cool to surface these as suggested prompts (as a future feature)

**Review by nlarew — APPROVED** [review]
LGTM, would love to do some eval-driven fine tuning on different aspects of this system prompt at some point in the future

File: packages/chatbot-server-mongodb-public/src/systemPrompt.ts [hunk 3]
```diff
@@ -1,18 +1,23 @@
...
+
 NEVER include links in your answer.
 Format your responses using Markdown. DO NOT mention that your response is formatted in Markdown.
 If you include code snippets, use proper syntax, line spacing, and indentation.
+
+If you include a code example in your response, only include examples in one programming language,
```
[the remark this answers is not in the document]
Comment (mongodben) [reply packages/chatbot-server-mongodb-public/src/systemPrompt.ts:20]: i dont know how the system would handle that, but this was added to remove the possibility of generating examples for N different drivers, which was occurring w/ the gpt-4o+prompt eng changes

**Comment by mongodben** [comment]
> LGTM, would love to do some eval-driven fine tuning on different aspects of this system prompt at some point in the future

agree!

File: packages/chatbot-eval-mongodb-public/src/braintrust/evaluateRagConversationsData.ts (hunk 1 above)
Comment (mongodben) [reply packages/chatbot-eval-mongodb-public/src/braintrust/evaluateRagConversationsData.ts:32]: funny that this works. will fix before merge

**Pull request merged**